packages/server/src/lib/access-helpers.js

```diff
@@ -1,19 +1,19 @@
-const { getUser } = require('../db');
-
-function isPartOfAgency(agencies, agencyId) {
-    return agencies.find((s) => s.id === Number(agencyId));
-}
+const { getUser, inTenant } = require('../db');
 
 /**
  * Determine if a user is authorized for an agency.
  *
- * @param {Number} userId
+ * @param {Object} user
  * @param {Number} agencyId
- * @returns {Boolean} true if the agency is the user's or a descendant; false otherwise
- */
+ * @returns {Boolean} true if the agency is in the same tenant as the user
+ * */
+async function isUserAuthorized(user, agencyId) {
+    return inTenant(user.id, user.tenant_id, agencyId);
```

Comment:
Good point, I think I missed this nuance after removing the super tenant concept.

Reply:
Huh? So what are the semantics _supposed to be_?

Reply:
Impersonation is limited to children in your agency tree, which is important when checking the selectedAgency, not just when choosing an agency to impersonate. In almost all other situations users can see objects across all agencies in a tenant. Like the user listing api is not limited by membership in a child agency anymore. So I'll be add a different function that checks for allowable impersonation vs allowing access to a resource... OR ask @mhuang-usdr If its okay if we let admin users (which are the only ones that can impersonate) impersonate any agency in a tenant, to help simplify things.

Reply:
Got it. I suspect the answer is we want impersonation to still be subagencies only.

Reply:
Clarifying question on what "allowable impersonation vs allowing access to a resource" means -

Let's say we have this agency tree:

- main agency
   -  A
      - sub-A
   - B

So,  if an admin in Agency A is impersonates sub-A, they should have full write/edit access to sub-A.

**Question:**
- in this current PR, if an admin in Agency A is impersonates Agency B, what is the experience? Is it a read-only experience, or does Agency A have write access?

From the product perspective, read-only impersonation should be fine - what's important is that the user is not able to write to an agency they're not supposed to - so I'd lean towards the simplest engineering solution until we hear future feedback.

However, I'd like to test the user experience of such an impersonation to make sure it's not totally broken!

Reply:
@bspates can I check via the same ngrok link you sent earlier?

Reply:
Talked with @mhuang-usdr, the answer is that write access in all cases needs to be limited by the agency tree. The nuance here is that with multi-tenancy we have admin users able to see across agency trees in a tenant, BUT not edit them. This PR makes it so that admins can write/edit any object in the tenancy. The work to move to a read but not write across the tenancy is not small, so we're spinning that out to another issue, as this is more of a future use case than a current one.